_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -479,7 +523,13 @@ mfxStatus MFXVideoENCODE_Query(mfxSession session, mfxVideoParam *in, mfxVideoPa
 
             if (MFX_WRN_PARTIAL_ACCELERATION == mfxRes)
             {
+                assert(handler != codecId2Handlers.end());
+#if defined(MFX_ENABLE_SW_FALLBACK)
```

Comment:
You don't need ifdef here. If MFX_ENABLE_SW_FALLBACK is not defined handler->second.fallback.query will be true and you correctly return MFX_ERR_UNSUPPORTED via runtime check.